sdk/webpubsub/Microsoft.Azure.WebPubSub.AspNetCore/src/Extensions/ServiceResponseExtensions.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Azure.WebPubSub.AspNetCore
+{
+    /// <summary>
+    /// Help methods for ServiceResponse.
+    /// </summary>
+    public static class ServiceResponseExtensions
+    {
+        /// <summary>
+        /// Set connection states.
+        /// </summary>
+        /// <param name="response"></param>
+        /// <param name="key"></param>
+        /// <param name="value"></param>
+        public static void SetState(this ServiceResponse response, string key, object value)
+        {
+            // value exists.
+            if (response.States.TryGetValue(key, out var value1))
+            {
+                response.States[key] = value;
+            }
+            // value new.
```

Comment:
Is there a difference between the 2 cases?
For dictionaries, I believe `response.States[key] = value;` will cover both cases - add the key-value pair if the key does not exist and overwrite with the new value if the key already exists.